packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js

```diff
@@ -75,106 +88,102 @@ class CloudFormation {
           running in the background */
         } else {
           let stackEvents = data.StackEvents;
-          stackEvents = stackEvents.filter(event => self.eventStartTime
-            < new Date(event.Timestamp));
+          stackEvents = stackEvents
+            .filter(event => self.eventStartTime < new Date(event.Timestamp));
 
-          if (allShownEvents.length === 0) { // Showing events for the first time to the user
+          if (allShownEvents.length === 0) {
+            // Showing events for the first time to the user
             if (stackEvents.length > 0) {
               showEvents(stackEvents);
               allShownEvents = stackEvents;
             }
-
-            // To store all the uniq nestedStacks that are a part of the new found events
-            const nestedStacks = [];
-            const nestedStackEventPromises = [];
-            let newEvents = stackEvents;
-
-            for (let i = 0; i < newEvents.length; i += 1) {
-              if (!nestedStacks.includes(newEvents[i].PhysicalResourceId)) {
-                // Logic to fetch events from nested stack within the parent stacks
-                if (newEvents[i].PhysicalResourceId.includes('arn:aws:cloudformation')) {
-                  /*eslint-disable*/
-                  nestedStackEventPromises.push(self.getNestedStackEvents(newEvents[i].PhysicalResourceId));
-                  /* eslint-enable */
-                  nestedStacks.push(newEvents[i].PhysicalResourceId);
-                }
-              }
-            }
-
-            return Promise.all(nestedStackEventPromises)
-              .then((nestedstackEvents) => {
-                newEvents = nestedstackEvents.reduce((combinedEventList, nestedstackEventList) =>
-                  combinedEventList.concat(nestedstackEventList), newEvents);
-                // Just get the new events to display to the user
-
-                newEvents = _.differenceBy(newEvents, allShownEvents, 'EventId');
-                newEvents = _.uniqBy(newEvents, 'EventId');
-                showEvents(newEvents);
-                allShownEvents = allShownEvents.concat(newEvents);
-              });
           }
 
           let newEvents = stackEvents;
```

Comment:
Shouldn't there be an else out here?

Reply:
We don't need else block as we dedupe the events when showing